addons/loyalty/views/res_partner_view.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record model="ir.ui.view" id="res_partner_view_buttons">
+            <field name="name">res.partner.view.buttons</field>
+            <field name="model">res.partner</field>
+            <field name="inherit_id" ref="base.view_partner_form" />
+            <field name="priority" eval="11"/>
+            <field name="arch" type="xml">
+                <div name="button_box" position="inside">
+                    <button class="oe_stat_button" type="object" name="action_view_loyalty_cards"
+                        groups="sales_team.group_sale_salesman"
```

Comment:
Since `loyalty` does not depend on `sales_team` we can not use that group.
I would suggest making another computed field that checks if the current user has read access on the loyalty card model and make the button invisible depending on that.
You will also have to check this field within the `loyalty_card_count` compute function.

Reply:
Actually we could also give it a system group (or a technical group such as `.`) 
```suggestion
                        groups="base.group_system"
```
And in `sale` (and `pos`) use an xpath with position attributes to add the groups we want to it, as the list will act as an or
```xml
<xpath expr="button[name=action_view_loyalty_card]" position="attributes">
    <attribute name="groups" separator="," add="sales_team.group_sale_salesman"/>
</xpath>
```